src/EhsnPlugin/Mappers/FieldVisitMapper.cs

```diff
@@ -156,60 +157,60 @@ private string MapWeather()
         {
             if (_eHsn.EnvCond == null) return null;
 
-            var lines = new List<string>();
+            var stringBuilder = new StringBuilder();
 
-            AddCommentLine(lines, "Cloud Cover", _eHsn.EnvCond.cloudCover);
-            AddCommentLine(lines, "Precipitation", _eHsn.EnvCond.precipitation);
-            AddCommentLine(lines, "Wind Conditions", _eHsn.EnvCond.windMagnitude);
-            AddCommentLine(lines, "Wind Speed", _eHsn.EnvCond.windMagnitudeSpeed);
-            AddCommentLine(lines, "Wind Direction", _eHsn.EnvCond.windDirection);
+            AddCommentLine(stringBuilder, "Cloud Cover", _eHsn.EnvCond.cloudCover);
+            AddCommentLine(stringBuilder, "Precipitation", _eHsn.EnvCond.precipitation);
+            AddCommentLine(stringBuilder, "Wind Conditions", _eHsn.EnvCond.windMagnitude);
+            AddCommentLine(stringBuilder, "Wind Speed", _eHsn.EnvCond.windMagnitudeSpeed);
+            AddCommentLine(stringBuilder, "Wind Direction", _eHsn.EnvCond.windDirection);
 
-            return string.Join("\n", lines);
+            return stringBuilder.ToString().Trim();
         }
 
-        private static void AddCommentLine(List<string> lines, string label, string value)
+        private static void AddCommentLine(StringBuilder stringBuilder, string label, string value)
         {
             if (string.IsNullOrWhiteSpace(value))
                 return;
 
-            var valueLines = value.Trim().Split('\n');
+            var trimmedValue = value.Trim();
 
-            if (valueLines.Length == 1)
+            // Single-line values map to a single line in the comment
+            if (trimmedValue.Count(v => v == '\n') == 0)
             {
-                // Single-line values map to a single line in the comment
-                lines.Add($"{label}: {valueLines[0]}");
+                stringBuilder.AppendFormat("{0}: {1}\n", label, trimmedValue);
```

Comment:
You could keep the more modern string formatting by switching to AppendLine:
```suggestion
                stringBuilder.AppendLine($"{label}: {trimmedValue}");
```
If that appends the wrong newline character, you could just fall back to Append:
```suggestion
                stringBuilder.Append($"{label}: {trimmedValue}\n");
```

Reply:
i intentionally stuck with `\n` instead of using `AppendLine` (which depends on the environment default line terminator) because i don't want any risk of changing parser behavior in this refactoring.  

also, wouldn't doing `stringBuilder.Append($"{label}: {trimmedValue}\n");` defeat the purpose of using stringbuilder?  string interpolation `$"{label}: {trimmedValue}\n"` would create a new string object, which is what i'm trying to avoid here.

Reply:
Using $ is not as inefficient as the original `string.Join()`, but you're right that the format would have to be evaluated before being appended to the StringBuilder, so would lose some efficiency.
Make sense to keep it as is.